ets-demo/etstool.py

```diff
@@ -311,6 +336,64 @@ def flake8(runtime, toolkit, environment):
     execute(commands, parameters)
 
 
+@cli.command(name="generate-bundles")
+def generate_bundles():
+    """ Generate application bundles for each platform to be uploaded to
+    enthought/free repository
+    """
+
+    for platform in PLATFORMS:
+        bundle_dir = os.path.join("bundle", platform)
+        if not os.path.exists(bundle_dir):
+            os.makedirs(bundle_dir)
+
+        bundle_file = os.path.join(
+            bundle_dir,
+            f"etsdemo_{platform.replace('-', '_')}-{APP_BUNDLE_VERSION}.bundle"
+        )
+
+        click.echo("Generating bundle {}".format(bundle_file))
+
+        edm_command = [
+            "edm",
+            "bundle",
+            "generate",
+            "--bundle-format",
+            "2.0",
+            "--platform",
+            platform,
+            "--version=3.6",
+            "--output-file",
+            bundle_file,
+        ] + sorted(full_app_dependencies)
+        try:
+            subprocess.check_call(edm_command)
+        except subprocess.CalledProcessError:
+            click.echo("Failed to generate bundle {}".format(bundle_file))
+
+
+@cli.command(name="upload-bundles")
+@click.option('--repository', default=DEFAULT_REPOSITORY)
+def upload_bundles(repository):
+    """ Upload the generated bundles """
+    BUNDLES = []
+    for platform in PLATFORMS:
+        bundle_dir = os.path.join("bundle", platform)
+        bundle_file = os.path.join(
+            bundle_dir,
+            f"etsdemo_{platform.replace('-', '_')}-{APP_BUNDLE_VERSION}.bundle"
+        )
+        if os.path.exists(bundle_file):
+            BUNDLES.append((bundle_file, platform))
+
+    for bundle, platform in BUNDLES:
+        cmd = "hatcher bundles upload enthought {repository} {platform} {bundle}"
+        execute(
```

Comment:
the use of `subprocess.check_call` vs `execute` when generate vs update bundles is slightly jarring. Is there a reason why we can't just use `subprocess.check_call` here? Is it more complicated/convoluted?